### ISSUE TITLE
Rectangle shape: grid based 'x' coordinate needs to be an integer

### ISSUE THREAD

**Issue by Looper1984** [goat1000/SVGGraph#53]
Hi, 

I stumbled upon  this weird behaviour: 
The x value in a rectangle needs to be an integer. Otherwise the placement of the rectangle will not be correct. 

I had to round the variable $gbx2 here: 

` require_once './SVGGraph/autoloader.php';
        $settings = [
          'auto_fit' => false,
          'back_colour' => '#fff',
          'back_stroke_width' => 0,
          'back_stroke_colour' => '#eee',
          'stroke_colour' => '#1089da',
          'line_stroke_width' => 2,
          'axis_colour' => '#444',
          'axis_overlap' => 0,
          'grid_colour' => '#bbb',
          'label_colour' => '#000',
          'axis_font' => 'Spartan',
          'axis_font_size' => 10,
          'fill_under' => false,
          'pad_right' => 0,
          'pad_left' => 0,
          'marker_type' => 'circle',
          'marker_size' => NULL,
          'marker_colour' => '#1089da',
          'link_base' => '/',
          'link_target' => '_top',
          'minimum_grid_spacing' => 20,
          'show_subdivisions' => true,
          'show_grid_subdivisions' => true,
          'grid_subdivision_colour' => '#eee',
          'label_font' => 'Spartan',
          'label_h' => 'Groove radius (mm)',
          'label_v' => 'Tracking distortion (%)',
          'axis_min_h' => 0,
          'axis_max_h' => 400,
          'axis_min_v' => 0,
          'axis_max_v' => 1,
          'shape' => [['rect', 'x'=>'g50', 'y'=>'g1', 'width'=>"u$bgw1", 'height'=>'u1', 'fill'=>'#ddd', 'stroke'=>'NULL', 'opacity'=>'50%'],
          ['rect', 'x'=>"g$bgx2", 'y'=>'g1', 'width'=>"u$bgw2", 'height'=>'u1', 'fill'=>'#ddd', 'stroke'=>'NULL', 'opacity'=>'50%']],

          
        ];`

**Comment by Looper1984** [comment]
Grid units with the u prefix work fine with non-integers, btw. The variable $bgw1 did not require rounding, so I guess this must be a bug

**Comment by goat1000** [comment]
Sorry it took me so long to look at this, GitHub didn't notify me until you posted your comment.

Non-integer values should be working - associative keys and datetime values can be used there as well. Are you using a locale where the decimal point is a comma or something?

**Comment by Looper1984** [comment]
No problem, when I first joined GitHub I used an email address that they didn't like and they flagged my account as 'spammy' - I guess that's why you didn't get a notification. 

Thanks for getting back to me :)
I do live in a country where the decimal point is a comma, but I don't think that's the problem here. The document is using decimal points exclusively.  I'm using non-integers for $bgw1 as well and that's working fine. I guess this problem may be related to the other issue I reported (https://github.com/goat1000/SVGGraph/issues/52)
